aioxmpp/avatar/service.py

```diff
@@ -340,9 +348,32 @@ def get_image_bytes(self):
         raise NotImplementedError
 
 
+class VCardAvatarDescriptor(AbstractAvatarDescriptor):
+
+    @property
+    def has_image_data_in_pubsub(self):
+        return False
+
+    @asyncio.coroutine
+    def get_image_bytes(self):
+        if self._image_bytes is not None:
+            return self._image_bytes
+
+        logger.debug("retrieving vCard %s", self._remote_jid)
+        vcard = yield from self._vcard.get_vcard(self._remote_jid)
+        photo = vcard.get_photo_data()
+        if photo is not None:
+            logger.debug("returning vCard avatar %s", self._remote_jid)
+            return photo
+
+        raise RuntimeError("Avatar image is not set")
+        # url = item.xpath("/ns0:PHOTO/ns0:EXTVAL/text()",
+        #                  namespaces={"ns0": namespaces.xep0054})
+
+
 class AvatarService(service.Service):
-    """
-    Access and publish User Avatars (:xep:`84`).
+    """Access and publish User Avatars (:xep:`84`). Fallback to vCard based
```

Comment:
FWIW, I know that PEP8 likes it, I don’t like the style where the comment starts right after ``"""``, and normally we’re not doing that in aioxmpp.

I don’t think it’s an issue though.

Reply:
I reconfigured my editor to handle this in the preferred (symmetric) style for aioxmpp.